### PR TITLE
765: Adding spider: Crate And Barrel

### LINKED ISSUE #765: Crate And Barrel
https://www.crateandbarrel.com/stores/list-state/retail-stores

### PR DESCRIPTION
**Pull request by kumanoit**
Fixes #765

### CODE REVIEW

File: locations/spiders/crateandbarrel.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+class CrateAndBarrelSpider(CrawlSpider):
+    name = "crateandbarrel"
+    allowed_domains = ["www.crateandbarrel.com"]
+    item_attributes = {"brand": "crateandbarrel", "brand_wikidata": "Q5182604"}
+    start_urls = ["https://www.crateandbarrel.com/stores/list-state/retail-stores"]
+    rules = [
+        Rule(
+            LinkExtractor(allow=r"stores\/list-state\/retail-stores\/([a-zA-Z]{2})$"),
+            callback="parse",
```
Comment (JoeDixon-TomTom) [inline locations/spiders/crateandbarrel.py:17]: Hi, I think if you changed the callback to "parse_sd" and inherited from StructuredDataSpider, bit like this one:
https://github.com/alltheplaces/alltheplaces/blob/master/locations/spiders/smashburger.py
.. then you would not need the parse() method below at all as parse_sd in the superclass would do it all for you. I hope so anyway! :-)
Cheers,
Joe.

Comment (kumanoit) [reply locations/spiders/crateandbarrel.py]: Thank you @JoeDixon-TomTom 
Worked like a charm :)

File: locations/spiders/davidstea.py [hunk 2]
```diff
@@ -1,7 +1,6 @@
 # -*- coding: utf-8 -*-
 from scrapy.spiders import SitemapSpider
 
-from locations.hours import OpeningHours
```
Comment (kumanoit) [inline locations/spiders/davidstea.py:4]: This was unused so removed it.
This spider is working well after removing above import.

File: locations/spiders/crateandbarrel.py (hunk 1 above)
Comment (Cj-Malone) [reply locations/spiders/crateandbarrel.py]: The issue now is that you're parsing the state URLs, and setting the ref it it's URL. So when a [state](https://www.crateandbarrel.com/stores/list-state/retail-stores/ca) has multiple stores all the stores get the same ref, and removed as duplicates.

```patch
-        Rule(
-            LinkExtractor(allow=r"stores\/list-state\/retail-stores\/([a-zA-Z]{2})$"),
-            callback="parse_sd",
-        )
+        Rule(
+            LinkExtractor(allow=r"stores\/list-state\/retail-stores\/([a-zA-Z]{2})$"),
+        ),
+        Rule(
+            LinkExtractor(allow=r"stores\/.+\/.+$"),
+            callback="parse_sd",
+        )
```

Comment (kumanoit) [reply locations/spiders/crateandbarrel.py]: Thank you..
changes incorporated

**Pull request merged**